db/openshift.deploy.yml

```diff
@@ -326,3 +329,95 @@ objects:
+                  args:
+                  - |
+                    sql_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql" -print0)
+                    if [  -z $sql_file ]; then
+                      sql_zip_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql.gz" -print0)
+                      if [ -z $sql_zip_file ]; then
+                        echo "No zipped backup file found"
+                      else
+                        echo "Found zipped backup file: $sql_zip_file"
+                        gzip -dk $sql_zip_file
+                      fi
+                    fi
+                    sql_file=$(find backups/${BACKUP_FILE_DIR} -type f -name "*.sql" -print0)
+                    if [  -z $sql_file ]; then
+                      echo "No backup SQL file is found"
+                    else
+                      echo "Found SQL file, rename existing database, and create a new empty database"
+                      psql -h ${NAME}-${ZONE}-${COMPONENT} -U ${POSTGRES_USER} -c "ALTER DATABASE fom RENAME TO ${RENAME_DATABASE_TO};" -c "CREATE DATABASE fom;"
+                      echo "Running SQL file: $sql_file"
+                      psql -h ${NAME}-${ZONE}-${COMPONENT} -U ${POSTGRES_USER} -d ${POSTGRES_DB} -f $sql_file
+                      echo "Finish database restore"
+                    fi
+                  volumeMounts:
+                    - name: ${NAME}-${ZONE}-${COMPONENT}-restore
+                      mountPath: "${BACKUP_DIR}"
+                  env:
+                    - name: BACKUP_DIR
+                      value: "${BACKUP_DIR}"
+                    - name: POSTGRES_DB
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-db
+                    # this enviornment variable mmust be named PGPASSWORD, so postgres knows this is the password, and will not ask us to enter manually
+                    - name: PGPASSWORD
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-password
+                    - name: POSTGRES_USER
+                      valueFrom:
+                        secretKeyRef:
+                          name: ${NAME}-${ZONE}-${COMPONENT}
+                          key: database-user
+                    # we don't need to declare the BACKUP_FILE_DIR variable if we set it manually, just put here so we won't forget
+                    # BACKUP_FILE_DIR need to be set manually, in the format for example, daily/2024-08-16, where backup container puts the backup files
+                    - name: BACKUP_FILE_DIR
+                      value: ""
+                    # RENAME_DATABASE_TO need to be set manually, rename the existing fom database to a new name
+                    - name: RENAME_DATABASE_TO
```

Comment:
Hi Ian, how we want to name this variable? basically this will be the name that we rename the existing "fom" database to.... I don't know what it should be called...

Reply:
Hi Catherine, I have no preference for the name. Does fom_old sound good?

Reply:
how about OLD_FOM_DATABASE_NAME?

Reply:
I think this should have a default value...

Reply:
(so it doesn't need to be specified every time).

Reply:
yeah I was using a default value before. And while doing testing, if I want to run the restore multiple times, I need to manually remove the renamed database, otherwise I'll got the error while renaming the database again, cause it already exists from the previous restore. So I was thinking maybe we can set the name manually, but I'm fine with a default value.